File: app/src/main/java/com/diamond/appcliente/dto/common/ApiResponse.java
package com.diamond.appcliente.dto.common;

public class ApiResponse<T> {

    private String status;
    private String message;
    private T data;

    public ApiResponse(String status, String message, T data) {
        this.status = status;
        this.message = message;
        this.data = data;
    }

    // Getters y setters
    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    // Método estático para crear una respuesta exitosa
    public static <T> ApiResponse<T> success(String message, T data) {
        return new ApiResponse<>("success", message, data);
    }

    // Método estático para crear una respuesta de error
    public static <T> ApiResponse<T> error(String message, T data) {
        return new ApiResponse<>("error", message, data);
    }
}

File: app/src/main/java/com/diamond/appcliente/dto/common/ApiResponse.kt
package com.diamond.appcliente.dto.common

class ApiResponse<T>(
    val status: String,
    val message: String,
    val data: T?
) {
    companion object {
        fun <T> success(message: String, data: T?) = ApiResponse("success", message, data)
        fun <T> error(message: String, data: T?) = ApiResponse("error", message, data)
    }
}
